lib/rust/ensogl/core/src/display/object/instance.rs

```diff
@@ -1020,139 +1489,1173 @@ impl InstanceDef {
+            }
+        }
+    }
+
+    fn refresh_manual_layout<Dim>(&self, x: Dim, pass: LayoutResolutionPass, pass_cfg: PassConfig)
+    where Dim: ResolutionDim {
+        let hug_children = pass_cfg != PassConfig::DoNotHugDirectChildren;
+        let hug_children = hug_children && self.layout.size.get_dim(x).is_hug();
+        let children = self.children();
+
+        let mut min_x = if children.is_empty() { 0.0 } else { f32::MAX };
+        let mut max_x = if children.is_empty() { 0.0 } else { f32::MIN };
+        let mut children_to_grow = vec![];
+        for child in &children {
+            if child.layout.grow_factor.get_dim(x) > 0.0 {
+                children_to_grow.push(child);
+            } else {
+                child.reset_size_to_static_values(x, self.layout.computed_size.get_dim(x));
+                child.refresh_layout_internal(pass, PassConfig::Default);
+                let child_pos = child.position().get_dim(x);
+                let child_size = child.computed_size().get_dim(x);
+                let child_content_origin = child.content_origin().get_dim(x);
+                let child_min_x = child_pos + child_content_origin;
+                let child_max_x = child_min_x + child_size;
+                min_x = min_x.min(child_min_x);
+                max_x = max_x.max(child_max_x);
+            }
+        }
+        if hug_children {
+            self.layout.computed_size.set_dim(x, max_x - min_x);
+        }
+        for child in children_to_grow {
+            child.layout.computed_size.set_dim(x, self.layout.computed_size.get_dim(x));
+            child.refresh_layout_internal(pass, PassConfig::DoNotHugDirectChildren);
+            let child_pos = child.position().get_dim(x);
+            let child_content_origin = child.content_origin().get_dim(x);
+            min_x = min_x.min(child_pos + child_content_origin);
+        }
+        self.layout.content_origin.set_dim(x, min_x);
+
+        let self_size = self.layout.computed_size.get_dim(x);
+        let forced_origin_alignment = self.layout.forced_origin_alignment.get().get_dim(x);
+        let origin_shift = forced_origin_alignment.normalized() * self_size;
+        self.layout.content_origin.update_dim(x, |t| t - origin_shift);
+    }
+}
+
+
+
+#[derive(Debug, Deref, DerefMut)]
+pub struct UnresolvedColumn {
+    #[deref]
+    #[deref_mut]
+    axis:     ColumnOrRow,
+    children: Vec<Instance>,
+}
+
+#[derive(Debug, Deref, DerefMut)]
+pub struct ResolvedColumn {
+    #[deref]
+    #[deref_mut]
+    axis:     ResolvedAxis,
+    children: Vec<Instance>,
+}
+
+#[derive(Clone, Copy, Debug, Default, PartialEq)]
+pub struct ResolvedAxis {
+    size:              Size,
+    min_size:          f32,
+    max_size:          f32,
+    grow_factor:       f32,
+    shrink_factor:     f32,
+    computed_size:     f32,
+    max_child_fr_size: Fraction,
+}
+
+impl Model {
+    fn divide_children_to_columns<Dim>(
+        &self,
+        x: Dim,
+        opts: &AutoLayout,
```

Comment:
So, the column in this place is actually a column o row? I just want to make sure. Actually, replacing "column" with "column or row" might be actually much more readable, even if a bit noisy. Looking at this code right now, I don't know where to look for information, that the column might be actually a row.

Reply:
Added explanation